### DISCUSSION TITLE
Process crashed. Simulation process failed. Exited with code 0xffffffffc0000005.

### DISCUSSION THREAD

**Discussion by bflo96** [OpenModelica/OpenModelica#13273] [Q&A]
Hi everyone!

I am creating a model in OpenModelica v1.23.1 (64-bit).

I initially created it with a smaller dimension to be able to track the variables and equations more easily. 
It is extremely non-linear. It has fairly simple components (resistors, current sources, and grounds).
The values of the resistances is calculated through functions, which use other variables of the system. 

It has a total of 498 equations/variables, which I could solve it after using LOG_NLS.

When I try to increase the size and/or change the current at the current source to a smaller/larger value, with the maximum size tried being 2084 equations/variables, it crashes:

> Process crashed.
>  Simulation process failed. Exited with code 0xffffffffc0000005.

With some logging, I was able to get more info. This happens for the majority, if not all, of the variables.

> read for Model.array[1,3] negated 0 from setup file

What does this error means? I tried to research about it, but I couldn't find anything in particular.

I am aware it might be difficult to get some help without sharing the code, but unfortunately I can't do it. I tried to do my best to explain the non-linearity of the problem and its complexity, so I am still hoping this could be enough.

Thank you in advance!

**Comment by casella** [comment]
Before doing anything else, please try with the recently released v. 1.24.2. Thanks!

**Reply by bflo96** [reply]
I just installed the new version and attempted to simulate my model, but the problem still remains. 
It crashes almost immediately after compiling.

**Reply by adrpo** [reply]
There are 3 ways to handle this:
- you send the model to our email (OpenModelica@ida.liu.se) and we delete it after we debug with it
- you send us the model executable (a zip of the entire directory would be good, as dependent .xml and .json files are needed to run it)
- we attempt to tell you how to debug the model locally and you send us the info with the crash

Let us know how you want to proceed.

**Reply by bflo96** [reply]
Hi! Sorry for the late response.
Could we attempt the 3rd option? How do I correctly debug the model locally?

Also, I am not sure if it is something that can give any hint regarding the non-linearity and complexity of the system I'm modeling, but in my equations I am analyzing if a current is positive and negative and using it to define other variables. Something like this:

```
if (var1.i > 0 and var2.i > 0) then
   var3 = (etc) * var1.i;
elseif (var1.i < 0 and var2.i > 0) then
   var3 = (etc) * var1.i; 
else
   var3 = (etc) * var1.i;
end if;
```

Is this something impossible for OM to solve? Could it be the reason why the process crashes?

Thank you!

**Reply by casella** [reply]
In general, it should. Given the structure of your model, you can try turning it into an equation with conditional expression:

```modelica
var3 = (if var1.1 > 0 and var2.i > 0 then (etc) else if var1.i < 0 and var2.i > 0 then (etc) else (etc)) * var1.i
```

**Reply by casella** [reply]
Maybe as the size goes up, avoiding the conditional equations helps
